File: AcademicPlanBackend/src/main/java/com/bntu/fitr/poit/zholudev/diplom/entity/DiagnosticTool.java
package com.bntu.fitr.poit.zholudev.diplom.entity;

import javax.persistence.*;
import javax.persistence.Entity;

@Entity
@Table(name = "diagnostic_tools")
public class DiagnosticTool {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String text;

    public DiagnosticTool() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }
}

File: AcademicPlanFrontend/src/app/entity/diagnostic-tool.ts
export class DiagnosticTool {
  id: number;
  text: string;

  static clone(diagnosticTool: DiagnosticTool): DiagnosticTool{
    let clonedDiagnosticTool = new DiagnosticTool();
    clonedDiagnosticTool.id = diagnosticTool.id;
    clonedDiagnosticTool.text = diagnosticTool.text;
    return clonedDiagnosticTool;
  }
}
